File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& v, int k) {
        unordered_map<int,int> mp;
        mp[0]=1;
        int s=v[0],ss=v[0];
        for(int i=1;i<v.size();i++){
            s+=v[i];
            int rem=s%k;
            if(rem==0 || mp.count(rem))return true;
            mp[ss%k]++;
            ss=s;
        }
        return false;
    }
};

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.java
class Solution {
    public boolean checkSubarraySum(int[] v, int k) {
        int s = v[0];
        Set<Integer> st = new HashSet<>();
        st.add(0);
        for(int i = 1; i < v.length; i += 1) {
            s += v[i];
            int rem = s % k;
            if(st.contains(rem)) return true;
            st.add((s - v[i]) % k);
        }
        return false;
    }
}